sansa-inference-spark/src/main/scala/net/sansa_stack/inference/spark/forwardchaining/axioms/ForwardRuleReasonerRDFS.scala

```diff
@@ -17,10 +27,383 @@ class ForwardRuleReasonerRDFS(sc: SparkContext, parallelism: Int = 2) extends Lo
   def apply(sc: SparkContext, parallelism: Int = 2): ForwardRuleReasonerRDFS =
     new ForwardRuleReasonerRDFS(sc, parallelism)
 
-  /**
-    * TODO: To be implemented
+
+  def apply(axioms: RDD[OWLAxiom], input: String): Unit = {   // : RDD[OWLAxiom]
+
+    val owlFile: File = new File(input)
+
+    val manager = OWLManager.createOWLOntologyManager()
+
+    val ontology: OWLOntology = manager.loadOntologyFromOntologyDocument(owlFile)
+
+    val dataFactory = manager.getOWLDataFactory
+
+    val axiomsRDD = axioms.cache()    // cache this RDD because it will be used quiet often
+
+    // ------------ extract the schema data -------------------
+    // Schema classes
+
+    val classes: RDD[OWLClass] = axiomsRDD.flatMap {
+      case axiom: HasClassesInSignature => axiom.classesInSignature().iterator().asScala
+      case _ => null
+    }.filter(_ != null).distinct()
+
+    println("\n\nOWL Classes\n-------\n")
+    classes.collect().foreach(println)
+
+    // OWLClassAssertionAxiom
+    val classAsserAxiom = axiomsRDD
+      .filter(axiom => axiom.getAxiomType.equals(AxiomType.CLASS_ASSERTION)).asInstanceOf[RDD[OWLClassAssertionAxiom]].cache()
+
+//    val cmap: Map[OWLClassExpression, Set[OWLIndividual]] = CollectionUtils
+//      .toMultiMap(classAsserAxiom.asInstanceOf[RDD[OWLClassAssertionAxiom]]
+//        .map(a => (a.getClassExpression, a.getIndividual)).collect())
+//    val c: Broadcast[Map[OWLClassExpression, Set[OWLIndividual]]] = sc.broadcast(cmap)
+
+//    val r = new Reasoner(axioms)
+//    val b: RDD[OWLClassAssertionAxiom] = classAsserAxiom.filter(a => r.isEntailed(a))
+//    println("\n\nb\n-------\n")
+//
+//    b.take(b.count().toInt).foreach(println(_))
+
+    println("\n\nOWLClassAssertionAxioms\n-------\n")
+    classAsserAxiom.collect().foreach(println)
```

Comment:
Please remove those debug `println`s and even more importantly those `.collect( )` calls.